frontend/src/tests/application-tests.js

```diff
@@ -1,19 +1,116 @@
-import { it, beforeAll } from "./utils";
+import {
+    it,
+    beforeAll,
+    expect,
+    checkPropTypes,
+    postingPropTypes,
+} from "./utils";
 import { databaseSeeder } from "./setup";
 
 export function applicationsTests({ apiGET, apiPOST }) {
-    beforeAll(async () => {
-        await databaseSeeder.seed({ apiGET, apiPOST });
-    }, 30000);
+    let session;
+    let position;
+    let posting = {};
+    let resp;
+    let admin;
+    let surveyData;
+
+    const taOnlyUser = {
+        utorid: "matthewc",
+        roles: ["ta"],
+    };
+
+    const postingData = {
+        name: "2021 Summer Posting",
+        intro_text: "Intro text for posting",
+        open_date: new Date("2021/05/01").toISOString(),
+        close_date: new Date("2021/08/31").toISOString(),
+        availability: "auto",
+    };
 
     // These tests set data through the `/public/postings` route,
     // but read data through the `/api/v1/admin` route.
     describe("Public route tests", () => {
+        beforeAll(async () => {
+            await databaseSeeder.seed({ apiGET, apiPOST });
+            session = databaseSeeder.seededData.session;
+            position = databaseSeeder.seededData.position;
+            resp = await apiGET(`/debug/active_user`);
+            expect(resp).toHaveStatus("success");
+            admin = {
+                utorid: resp.payload.utorid,
+                roles: resp.payload.roles,
+            };
+        });
+
         it.todo("Get survey.js posting data through public route");
         it.todo(
             "Survey.js posting data is pre-filled based on prior applicant/application data"
         );
-        it.todo("Can submit survey.js data via the public postings route");
+
+        it("Can submit survey.js data via the public postings route", async () => {
+            //Create a new posting
+            resp = await apiPOST(
+                `/admin/sessions/${session.id}/postings`,
+                postingData
+            );
+            expect(resp).toHaveStatus("success");
+            Object.assign(posting, resp.payload);
+            checkPropTypes(postingPropTypes, posting);
+            expect(posting.id).not.toBeNull();
+
+            //Set position for posting
+            resp = await apiPOST(
+                `/admin/postings/${posting.id}/posting_positions`,
+                { hours: 20, num_positions: 2, position_id: position.id }
```

Comment:
Setting these fields is fine. You could also leave them out, because they have default values.